templates/install.j2

```diff
@@ -71,40 +59,117 @@ ynh_setup_source --dest_dir="$install_dir"
 {% if data.main_technology == "php" -%}
-# Create a dedicated PHP-FPM config
-# conf/extra_php-fpm.conf will be appended to the auto-generated config, which will go in /etc/php/X.Y/fpm/pool.d/
-ynh_add_fpm_config
+ynh_config_add_phpfpm
 {% endif %}
 
-# Create a dedicated NGINX config using the conf/nginx.conf template
-ynh_add_nginx_config
+ynh_config_add_nginx
 
 {% if data.main_technology not in ["php", "none"] -%}
 {% if data.generator_mode == 'tutorial' -%}
-### `ynh_systemd_config` is used to configure a systemd script for an app, using the conf/systemd.service template
-{% endif %}
-# Create a dedicated systemd config
-ynh_add_systemd_config
+### `ynh_config_add_systemd` is used to configure a systemd script for an app.
+### It can be used for apps that use sysvinit (with adaptation) or systemd.
+### Have a look at the app to be sure this app needs a systemd script.
+### `ynh_config_add_systemd` will use the file conf/systemd.service
+### If you're not using these lines:
+###   - You can remove those files in conf/.
+###   - Remove the section "BACKUP SYSTEMD" in the backup script
+###   - Remove also the section "STOP AND REMOVE SERVICE" in the remove script
+###   - As well as the section "RESTORE SYSTEMD" in the restore script
+###   - And the section "SETUP SYSTEMD" in the upgrade script
+{% endif %}
+ynh_config_add_systemd
 
 {% if data.generator_mode == 'tutorial' -%}
 ### `yunohost service add` integrates a service in YunoHost. It then gets
 ### displayed in the admin interface and through the others `yunohost service` commands.
-{% endif %}
+### (N.B.: this line only makes sense if the app adds a service to the system!)
+### If you're not using these lines:
+###   - You can remove these files in conf/.
+###   - Remove the section "REMOVE SERVICE INTEGRATION IN YUNOHOST" in the remove script
+###   - As well as the section "INTEGRATE SERVICE IN YUNOHOST" in the restore script
+###   - And the section "INTEGRATE SERVICE IN YUNOHOST" in the upgrade script
 
-yunohost service add $app --log="/var/log/$app/$app.log"
+### Additional options starting with 3.8:
+###
+### --needs_exposed_ports "$port" a list of ports that needs to be publicly exposed
+###                               which will then be checked by YunoHost's diagnosis system
+###                               (N.B. DO NOT USE THIS if the port is only internal!!!)
+###
+### --test_status "some command"  a custom command to check the status of the service
+###                               (only relevant if 'systemctl status' doesn't do a good job)
+###
+### --test_conf "some command"    some command similar to "nginx -t" that validates the conf of the service
+###
+### Re-calling 'yunohost service add' during the upgrade script is the right way
+### to proceed if you later realize that you need to enable some flags that
+### weren't enabled on old installs (be careful it'll override the existing
+### service though so you should re-provide all relevant flags when doing so)
```

Comment:
Eeeh, this comment block is already there right after the call to `service add` ?